dev/generate_protos.py

```diff
@@ -0,0 +1,203 @@
+    (
+        "import databricks_uc_registry_messages_pb2 as databricks__uc__registry__messages__pb2",
+        "from . import databricks_uc_registry_messages_pb2 as databricks_uc_registry_messages_pb2",
+    ),
+    (
+        "import databricks_managed_catalog_messages_pb2 as databricks__managed__catalog__"
+        "messages__pb2",
+        "from . import databricks_managed_catalog_messages_pb2 as databricks_managed_"
+        "catalog_messages_pb2",
+    ),
+]
+
+
+def gen_python_protos(protoc_bin, protoc_include_path, out_dir):
+    gen_protos(
+        mlflow_protos_dir, python_proto_files, "python", protoc_bin, protoc_include_path, out_dir
+    )
+
+    gen_protos(
+        test_protos_dir, test_proto_files, "python", protoc_bin, protoc_include_path, out_dir
+    )
+
+    for file_name in python_proto_files:
+        apply_python_gencode_replacement(Path(out_dir, _get_python_output_filename(file_name)))
+
+
+def download_and_extract_protoc(version):
+    """
+    Download and extract specific version protoc tool, return extracted protoc executable file path
+    and include path.
+    """
+    assert platform.system() in [
+        "Darwin",
+        "Linux",
+    ], "The script only supports MacOS or Linux system."
+    assert platform.machine() in [
+        "x86_64",
+        "aarch64",
+    ], "The script only supports x86_64 or aarch64 CPU."
+
+    os_type = "osx" if platform.system() == "Darwin" else "linux"
+    cpu_type = "x86_64" if platform.machine() == "x86_64" else "aarch_64"
+
+    downloaded_protoc_bin = f"{cache_dir}/protoc-{version}/bin/protoc"
+    downloaded_protoc_include_path = f"{cache_dir}/protoc-{version}/include"
+
+    protoc_zip_filename = f"protoc-{version}-{os_type}-{cpu_type}.zip"
+    if not (
+        os.path.isfile(downloaded_protoc_bin) and os.path.isdir(downloaded_protoc_include_path)
+    ):
+        subprocess.check_call(
+            [
+                "wget",
+                f"https://github.com/protocolbuffers/protobuf/releases/download/v{version}/{protoc_zip_filename}",
+                "-O",
+                f"{cache_dir}/{protoc_zip_filename}",
+            ]
+        )
+        subprocess.check_call(
+            [
+                "unzip",
+                "-o",
+                "-d",
+                f"{cache_dir}/protoc-{version}",
+                f"{cache_dir}/{protoc_zip_filename}",
+            ]
+        )
+    return downloaded_protoc_bin, downloaded_protoc_include_path
+
+
+def prepend_intent(content):
+    lines = content.split("\n")
+    lines = ["  " + line for line in lines]
+    return "\n".join(lines)
+
+
+def generate_final_python_gencode(gencode3194_path, gencode5260_path, out_path):
+    gencode3194 = gencode3194_path.read_text()
+    gencode5260 = gencode5260_path.read_text()
+
+    merged_code = f"""
+import google.protobuf
+from packaging.version import Version
+if Version(google.protobuf.__version__) >= Version("5.26.0"):
```

Comment:
```suggestion
if Version(google.protobuf.__version__).major >= 5:
```

can we just compare the major version?